### PR TITLE
Support domain type concept exclusions.

### PR DESCRIPTION
**Pull request by chrisknoll**
Adds a new UI element to toggle 'is any of' and 'is not any of' for domain type concepts. This allows you to do things like 'where type not in patient reported'.

### CODE REVIEW

File: js/modules/cohortbuilder/components/ConditionOccurrenceTemplate.html [hunk 1]
```diff
@@ -49,7 +49,7 @@
 			<tr data-bind="if: ConditionType() != null, visible: ConditionType() != null">
 				<td><i data-bind="click: function() { $component.removeCriterion('ConditionType') }" class="fa fa-times" /></td>
 				<td>
-					Condition Type is:
+					Condition Type <button class="btn btn-primary btn-xs" data-bind="click: function() { ConditionTypeExclude(!ConditionTypeExclude())}, text: ConditionTypeExclude() ? 'is not any of': 'is any of'"></button>
```
Comment (pavgra) [inline js/modules/cohortbuilder/components/ConditionOccurrenceTemplate.html]: The added button is pretty the same in all changed files: wouldn't it be reasonable to move it into a separate template which accepts boolean observable?

Comment (chrisknoll) [reply js/modules/cohortbuilder/components/ConditionOccurrenceTemplate.html]: We can do that.

**Comment by chrisknoll** [comment]
@pavgra : I've moved the duplicated code into a new Input component: CycleToggleInput.

File: js/components/cohortbuilder/components/CycleToggleInput.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+		
+		self.value = params.value;
+		self.options = params.options;
+		
+		self.currentIndex = ko.pureComputed(() => getOptionIndex(self.value(), self.options));
+		self.selectedOptionText = ko.pureComputed(function() {
+			return self.options[self.currentIndex()].name;
+		});
+		
+		self.nextOption = function () {
```
Comment (pavgra) [inline js/components/cohortbuilder/components/CycleToggleInput.js]: Looks too similar to `js/components/cohortdefinitionviewer/components/CycleToggleInput.js`. Why not to move common parts out / have single parametrized component?

**Pull request merged**